### ISSUE TITLE
baseline command issue with Redshift

### ISSUE THREAD

**Issue by alexhajjat** [flyway/flyway#1041]
I found a bug with “Flyway baseline” command in Redshift and want to report it to you 
 
Creating Metadata table: "public"."schema_version"
ERROR: Unable to insert row for version '1' in metadata table "public"."schema_version"

**Comment by cajnoj** [comment]
It's trying to create the metadata table in the public schema. Maybe you don't have permissions to do so? What is your "schemas" parameter set to? And try running it with -X in order to get the exact error that flyway encounters.

**Comment by alexhajjat** [comment]
DEBUG: Executing SQL: ALTER TABLE "public"."schema_version" ADD CONSTRAINT "schema_version_pk" PRIMARY KEY ("version")
DEBUG: Metadata table "public"."schema_version" created.
DEBUG: Rolling back transaction...
DEBUG: Transaction rolled back
ERROR: Unexpected error
org.flywaydb.core.api.FlywayException: Unable to insert row for version '1' in metadata table "public"."schema_version"
        at org.flywaydb.core.internal.metadatatable.MetaDataTableImpl.addAppliedMigration(MetaDataTableImpl.java:140)
        at org.flywaydb.core.internal.metadatatable.MetaDataTableImpl.addBaselineMarker(MetaDataTableImpl.java:255)
        at org.flywaydb.core.internal.command.DbBaseline$2.doInTransaction(DbBaseline.java:116)
        at org.flywaydb.core.internal.command.DbBaseline$2.doInTransaction(DbBaseline.java:95)
        at org.flywaydb.core.internal.util.jdbc.TransactionTemplate.execute(TransactionTemplate.java:72)
        at org.flywaydb.core.internal.command.DbBaseline.baseline(DbBaseline.java:95)
        at org.flywaydb.core.Flyway$5.execute(Flyway.java:1186)
        at org.flywaydb.core.Flyway$5.execute(Flyway.java:1182)
        at org.flywaydb.core.Flyway.execute(Flyway.java:1418)
        at org.flywaydb.core.Flyway.baseline(Flyway.java:1182)
        at org.flywaydb.commandline.Main.executeOperation(Main.java:117)
        at org.flywaydb.commandline.Main.main(Main.java:91)
Caused by: java.sql.SQLDataException: [Amazon][JDBC](10120) Error converting data, invalid type for parameter: 7.
        at com.amazon.exceptions.ExceptionConverter.toSQLException(Unknown Source)
        at com.amazon.jdbc.common.SPreparedStatement.createConvertedParameterInputValue(Unknown Source)
        at com.amazon.jdbc.common.SPreparedStatement.getInputParameterValues(Unknown Source)
        at com.amazon.jdbc.common.SPreparedStatement.getInputParameterValues(Unknown Source)
        at com.amazon.jdbc.common.SPreparedStatement.executeUpdate(Unknown Source)
        at org.flywaydb.core.internal.dbsupport.JdbcTemplate.update(JdbcTemplate.java:271)
        at org.flywaydb.core.internal.metadatatable.MetaDataTableImpl.addAppliedMigration(MetaDataTableImpl.java:115)
        ... 11 more

**Comment by mrachelski** [comment]
I am getting the same error.

**Comment by mrachelski** [comment]
For others finding this, I finally got it to work by NOT using the redshift JDBC drivers. Instead, I used the following in my config file.

Do not specify a flyway.driver as this will then use the default drivers
Specify your redshift URL using the postgresql instead of the normal "redshift" after jdbc

An example looks like this:

flyway.url=jdbc:postgresql://connectionname.redshift.amazonaws.com:5439/somedb

This recipe allowed me to run flyway against redshift.
